nucliadb/src/nucliadb/common/external_index_providers/base.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+import abc
+import logging
+
+from nucliadb_protos.noderesources_pb2 import Resource
+
+logger = logging.getLogger(__name__)
+
+
+class ExternalIndexManager(abc.ABC, metaclass=abc.ABCMeta):
+    type: str
+
+    def __init__(self, kbid: str):
+        self.kbid = kbid
+
+    async def delete_resource(self, resource_uuid: str) -> None:
+        logger.info(
+            "Deleting resource to external index",
+            extra={
+                "kbid": self.kbid,
+                "resource_uuid": resource_uuid,
```

Comment:
I think we should start to standarize logging keys. We have a mix between `rid`, `resource_id` and `resource_uuid`. Same for the variable name (although less important IMO)